### DISCUSSION TITLE
Improper call to `setETCDLabelsAndAnnotations()`

### DISCUSSION THREAD

**Discussion by ckyoog** [k3s-io/k3s#4278] [Q&A]
https://github.com/k3s-io/k3s/blob/3f5774b41eb475eb10c93bb0ce58459a6f777c5f/pkg/server/server.go#L81-L85

Function `setETCDLabelsAndAnnotations()` creates a kube client (like kubectl) to do its job, it needs APIServer to run. So why call it when APIServer is disabled, which will fail for 100% sure.

It is fine though, the `--disable-apiserver` is experimental. P.S., I am looking at the tag [1.22.2+k3s2](https://github.com/k3s-io/k3s/tree/v1.22.2+k3s2).

**Comment by brandond** [comment]
Yes, this would not work if this was the only node in the cluster. However, this is meant to be used on an etcd-only server - one with all the Kubernetes control-plane components disabled. You'll notice that it's run in a goroutine so that it does not block. It will wait for another server node to come up and host the Kubernetes control-plane, at which point it will set the correct labels and annotations on this node.
 
 Are you running into some problem with this code, or just trying to understand the flow?

**Reply by ckyoog** [reply]
Thank you. I checked code because of some other issues. This one doesn't bother me, I just don't feel right about it. Thank you for explaining the background of this function. Some comments would be very helpful if it is possible.
